pkg/utils/k8sutil.go

```
@@ -847,3 +658,278 @@ func DeleteIngress(client kubernetes.Interface, name, ns string) error {
+	if err != nil && k8sErrors.IsAlreadyExists(err) {
+		data, _ := json.Marshal(dpm)
+		_, err = client.Extensions().Deployments(funcObj.Metadata.Namespace).Patch(dpm.Name, types.StrategicMergePatchType, data)
+		if err != nil {
+			return err
+		}
+
+		// kick existing function pods then it will be recreated
+		// with the new data mount from updated configmap.
+		// TODO: This is a workaround.  Do something better.
+		pods, err := GetPodsByLabel(client, funcObj.Metadata.Namespace, "function", funcObj.Metadata.Name)
+		for _, pod := range pods.Items {
+			err = client.Core().Pods(funcObj.Metadata.Namespace).Delete(pod.Name, &metav1.DeleteOptions{})
+			if err != nil && !k8sErrors.IsNotFound(err) {
+				// non-fatal
+				logrus.Warnf("Unable to delete pod %s/%s, may be running stale version of function: %v", funcObj.Metadata.Namespace, pod.Name, err)
+			}
+		}
+	}
+
+	return err
+}
+
+func ensureFuncJob(client kubernetes.Interface, funcObj *spec.Function, or []metav1.OwnerReference) error {
+	job := &v2alpha1.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            fmt.Sprintf("trigger-%s", funcObj.Metadata.Name),
+			Labels:          funcObj.Metadata.Labels,
+			OwnerReferences: or,
+		},
+		Spec: v2alpha1.CronJobSpec{
+			Schedule: funcObj.Spec.Schedule,
+			JobTemplate: v2alpha1.JobTemplateSpec{
+				Spec: batchv1.JobSpec{
+					Template: v1.PodTemplateSpec{
+						Spec: v1.PodSpec{
+							Containers: []v1.Container{
+								{
+									Image: busybox,
+									Name:  "trigger",
+									Args:  []string{"curl", fmt.Sprintf("http://%s.%s.svc:8080", funcObj.Metadata.Name, funcObj.Metadata.Namespace)},
+								},
+							},
+							RestartPolicy: v1.RestartPolicyOnFailure,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	_, err := client.BatchV2alpha1().CronJobs(funcObj.Metadata.Namespace).Create(job)
+	if err != nil && k8sErrors.IsAlreadyExists(err) {
+		data, _ := json.Marshal(job)
```

don't ignore errors here